libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/sound/SoundButtonComponent.kt

```diff
@@ -0,0 +1,44 @@
+package com.mapbox.navigation.dropin.component.sound
+
+import android.view.View
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.ui.voice.view.MapboxSoundButton
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class SoundButtonComponent(
+    private val soundButton: MapboxSoundButton,
+) : UIComponent() {
+
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        super.onAttached(mapboxNavigation)
+        val audioGuidance = MapboxNavigationApp.getObserver(MapboxAudioComponent::class)
+
+        coroutineScope.launch {
+            audioGuidance.stateFlow().collect { state ->
+                when (state.isMuted) {
+                    true -> soundButton.mute()
+                    else -> soundButton.unmute()
+                }
+            }
+        }
+        coroutineScope.launch {
+            soundButton.clicks().collect { audioGuidance.toggle() }
+        }
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private fun View.clicks() = callbackFlow {
+        setOnClickListener {
+            trySend(Unit)
+        }
+        awaitClose { setOnClickListener(null) }
+    }
+}
```

Comment:
Why overcomplicate this?  No need to have flowable for clicks. 

Just set and remove listener in `onAttached` and `onDetached`.

```kotlin
override fun onAttached(mapboxNavigation: MapboxNavigation) {
    super.onAttached(mapboxNavigation)
    val audioGuidance = MapboxNavigationApp.getObserver(MapboxAudioComponent::class)

    coroutineScope.launch {
        audioGuidance.stateFlow().collect { state ->
            if (state.isMuted) soundButton.mute()
            else soundButton.unmute()
        }
    }
    soundButton.setOnClickListener {
        audioGuidance.toggle()
    }
}

override fun onDetached(mapboxNavigation: MapboxNavigation) {
    super.onDetached(mapboxNavigation)
    soundButton.setOnClickListener(null)
}
```

Reply:
Yeah obvious alternative 😄 . The nice thing about flows, is you can combine them.

``` kotlin
combine(button.clicks(), stateFlow()) { view, state -> {
  view.background = state.background
}
```

We also, will not have to implement onDetached just to null out the click listener.

But, also have not decided a good place to put these types of extensions 🤔

Reply:
But not a big deal, removing it and will think about it later.